v2/component/launcher.go

```diff
@@ -275,16 +281,21 @@ func (l *Launcher) executeWithoutCacheEnabled(ctx context.Context, executorInput
 	cmd := l.cmdArgs[0]
 	args := make([]string, len(l.cmdArgs)-1)
 	_ = copy(args, l.cmdArgs[1:])
-	pipeline, err := l.metadataClient.GetPipeline(ctx, l.options.PipelineName, l.options.PipelineRunID)
+	pipeline, err := l.metadataClient.GetPipeline(ctx, l.options.PipelineName, l.options.RunID, l.options.Namespace, l.options.RunResource)
```

Comment:
Is it more clear if we pass l.options as the parameter?

Reply:
I intentionally did not do so, because metadata package is a low level package. It will be reused by both v2 compatible launcher and v2 driver/publisher.

It should not use a type defined in launcher package.